system/jlib/jlog.cpp

```diff
@@ -3482,69 +3482,78 @@ bool fetchLogByClass(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, Log
+            logAccessPluginConfig.setown(testTree->getPropTree("global/logAccess"));
+        }
 #endif
 
-    if (!logAccessPluginConfig)
-        throw makeStringException(-1, "RemoteLogAccessLoader: logaccess configuration not available!");
+        if (!logAccessPluginConfig)
+            throw makeStringException(-1, "RemoteLogAccessLoader: logaccess configuration not available!");
 
-    constexpr const char * methodName = "queryRemoteLogAccessor";
-    constexpr const char * instFactoryName = "createInstance";
+        constexpr const char * methodName = "queryRemoteLogAccessor";
+        constexpr const char * instFactoryName = "createInstance";
 
-    StringBuffer libName; //lib<type>logaccess.so
-    StringBuffer type;
-    logAccessPluginConfig->getProp("@type", type);
-    if (type.isEmpty())
-        throw makeStringExceptionV(-1, "%s RemoteLogAccess plugin kind not specified.", methodName);
-    libName.append("lib").append(type.str()).append("logaccess");
+        StringBuffer libName; //lib<type>logaccess.so
+        StringBuffer type;
+        logAccessPluginConfig->getProp("@type", type);
+        if (type.isEmpty())
+            throw makeStringExceptionV(-1, "%s RemoteLogAccess plugin kind not specified.", methodName);
+        libName.append("lib").append(type.str()).append("logaccess");
 
-    //Load the DLL/SO
-    HINSTANCE logAccessPluginLib = LoadSharedObject(libName.str(), true, false);
-    if(logAccessPluginLib == nullptr)
-        throw makeStringExceptionV(-1, "%s cannot load library '%s'", methodName, libName.str());
+        //Load the DLL/SO
+        HINSTANCE logAccessPluginLib = LoadSharedObject(libName.str(), true, false);
+        if(logAccessPluginLib == nullptr)
+            throw makeStringExceptionV(-1, "%s cannot load library '%s'", methodName, libName.str());
 
-    newLogAccessPluginMethod_t_ xproc = nullptr;
-    xproc = (newLogAccessPluginMethod_t_)GetSharedProcedure(logAccessPluginLib, instFactoryName);
-    if (xproc == nullptr)
-        throw makeStringExceptionV(-1, "%s cannot locate procedure %s in library '%s'", methodName, instFactoryName, libName.str());
+        newLogAccessPluginMethod_t_ xproc = nullptr;
+        xproc = (newLogAccessPluginMethod_t_)GetSharedProcedure(logAccessPluginLib, instFactoryName);
+        if (xproc == nullptr)
+            throw makeStringExceptionV(-1, "%s cannot locate procedure %s in library '%s'", methodName, instFactoryName, libName.str());
 
-    //Call logaccessplugin instance factory and return the new instance
-    DBGLOG("Calling '%s' in log access plugin '%s'", instFactoryName, libName.str());
-    IRemoteLogAccess * pLogAccessPlugin = dynamic_cast<IRemoteLogAccess*>(xproc(*logAccessPluginConfig));
+        //Call logaccessplugin instance factory and return the new instance
+        DBGLOG("Calling '%s' in log access plugin '%s'", instFactoryName, libName.str());
+        if (!logAccessor)
```

Comment:
If the 'CriticalBlock block(logAccessCrit);' is called before line 3490, you do not need to check the logAccessor here.